server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -69,9 +74,12 @@ public CoordinatorDynamicConfig(
       // coordinator console can not send array of strings in the update request.
       // See https://github.com/druid-io/druid/issues/3055
       @JsonProperty("killDataSourceWhitelist") Object killDataSourceWhitelist,
-      @JsonProperty("killPendingSegmentsSkipList") Object killPendingSegmentsSkipList,
       @JsonProperty("killAllDataSources") boolean killAllDataSources,
-      @JsonProperty("maxSegmentsInNodeLoadingQueue") int maxSegmentsInNodeLoadingQueue
+      @JsonProperty("killPendingSegmentsSkipList") Object killPendingSegmentsSkipList,
+      @JsonProperty("maxSegmentsInNodeLoadingQueue") int maxSegmentsInNodeLoadingQueue,
+      @JsonProperty("compactionConfigs") List<CoordinatorCompactionConfig> compactionConfigs,
```

Comment:
this needs to take Object for the same reason as that of killDataSourceWhitelist

Reply:
Hmm, ```CoordinatorDynamicConfigsResource``` provides an interface to enable/disable compaction for each dataSource like ```POST/GET/DELETE /druid/coordinator/v1/config/compaction/{dataSource}```. Maybe the coordinator console can use this APIs?

Reply:
independent of `/druid/coordinator/v1/config/compaction/{dataSource}` , compaction config can also be updated via general coordinator dynamic config endpoint `POST /druid/coordinator/v1/config`... and coordinator console already has ability to do that.
e.g. after this PR... compaction config would show up in coordinator console and will be update-able there but it would fail because console doesn't have ability to send json list.

Reply:
Hmm. I'm not much familiar with coffee script and node, but let's fix the coordinator console. Unlike ```killDataSourceWhitelist``` which allows a set of simple strings, ```compactionConfigs``` is a list of ```CoordinatorCompactionConfig``` each of which is a large JSON object. I think it's not a good UI if users should input configuration values as raw JSON objects.

Reply:
My intention wasn't really to change things on UI side and now I see your problem that its not a simple `List<String>`.
However this leads to a problem for users who would use auto compaction, they would not be able to update other coordinate dynamic configs from the console but will have to use manual POST requests for those too.

Reply:
My thinking is compaction config should be a different `druid_config` object, separate from the coordinator dynamic config. It doesn't really fit the coordinator dynamic config model which is meant to be a simple form that is filled out on the web console.

The compaction config is more complicated than that so I think it deserves its own config object. At first, it's ok if this is only editable via API.

Reply:
@jihoonson, @himanshug what do you think of the above?

Reply:
@gianm @himanshug ok. I'll make a new config object in a follow-up pr.

Reply:
@jihoonson that's ok, but this follow up pr must be done before a release including this feature (to avoid config migration issues) so please raise an issue with milestone 0.13.0 before this current pr is committed.

Reply:
Raised https://github.com/druid-io/druid/issues/5242.

Reply:
Thanks

Reply:
yep ... using separate space for storing compaction config sounds good to me. thanks.